### PR TITLE
Fixes for XML reporting

### PR DESCRIPTION
**Pull request by KKopyscinski**

### CODE REVIEW

**Comment by KKopyscinski** [comment]
@michal-narajowski PTAL

File: bot/common.py [hunk 1]
```diff
@@ -443,17 +443,22 @@ def make_report_xlsx(results_dict, status_dict, regressions_list,
...
 
     def find_xml_by_case(case):
-        nonlocal matched_xml
-        matched_xml = ''
-        to_match = case.replace('/', '_').replace('-', '_')
-        for xml in xml_list:
-            if to_match in xml.name:
-                matched_xml = xml.name
-                break
+        if xml_list is not None:
```
Comment (michal-narajowski) [inline bot/common.py:454]: Return early

**Review by michal-narajowski — APPROVED** [review]
LGTM, just a minor comment.

File: bot/common.py [hunk 2]
```diff
@@ -713,6 +722,14 @@ def recursive(directory, depth):
     return dir_path
 
 
+def parse_xml(file_path):
```
Comment (michal-narajowski) [inline bot/common.py:725]:
```suggestion
def split_xml_filename(file_path):
```

File: bot/common.py [hunk 3]
```diff
@@ -713,6 +722,14 @@ def recursive(directory, depth):
...
+    start_of_name = file_path.rfind('/')
+    filename = file_path[start_of_name + 1:]
```
Comment (michal-narajowski) [inline bot/common.py:727]:
```suggestion
    file_name = os.path.basename(file_path)
    file_name, _ = os.path.splitext(file_name)
    test_name, timestamp = file_name.split("_C_")
    return test_name, timestamp
```

File: bot/common.py [hunk 4]
```diff
@@ -761,6 +779,13 @@ def pull_server_logs(args):
 
                     if file_path.endswith('.xml') and not 'tc_log' in file_path\
                             and b'Final Verdict:PASS' in file_bin.data:
+                        (test_name, timestamp) = parse_xml(file_path)
+                        print(test_name, timestamp)
+                        if test_name in last_xml[0]:
+                            if timestamp > last_xml[1]:
+                                os.remove(last_xml[0])
+                            else:
+                                continue
```
Comment (michal-narajowski) [inline bot/common.py:788]:
```suggestion
                        (test_name, timestamp) = split_xml_filename(file_path)
                        if test_name in last_xml[0]:
                            if timestamp <= last_xml[1]:
                                continue
                            os.remove(last_xml[0])
                            
                                
```

**Review by michal-narajowski — CHANGES_REQUESTED** [review]

File: bot/common.py (hunk 4 above)
Comment (KKopyscinski) [reply bot/common.py]: This looks much cleaner :) I updated the PR

**Comment by KKopyscinski** [comment]
Solved issue with missing XMLs in report - `os.scandir()` returns an iterator object that removes it's elements when iterated through. Casting it to `list` soves the issue

**Review by michal-narajowski — APPROVED** [review]

**Pull request merged**